### PR TITLE
Add documenter docs

### PR DESCRIPTION
**Pull request by Roger-luo**
#83

### CODE REVIEW

**Comment by Roger-luo** [comment]
Used new documenter according: https://discourse.julialang.org/t/psa-use-a-project-for-building-your-docs/14974

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,19 @@
...
+docs/Manifest.toml
+
+*.ipynb_checkpoints
+**/*.ipynb_checkpoints
+**/**/*.ipynb_checkpoints
+
+_*.dat
+*.swp
+__pycache__/
+Manifest.toml
```
Comment (MikeInnes) [inline .gitignore]: I think it's ok to include these, actually. In future we'll be using them for CI anyway.

File: .travis.yml [hunk 2]
```diff
@@ -4,13 +4,28 @@ os:
...
 notifications:
   email: false
-# uncomment the following lines to override the default test script
-#script:
-#  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-#  - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.build("MacroTools"); Pkg.test("MacroTools"; coverage=true)'
+
+env:
+  global:
+    - DOCUMENTER_DEBUG=true
```
Comment (MikeInnes) [inline .travis.yml]: Is it right to leave this in when merging to master?

Comment (Roger-luo) [reply .travis.yml]: It depends on what you prefer. The configuration on discourse has the debug option on.

**Comment by MikeInnes** [comment]
Is this still WIP?

**Comment by Roger-luo** [comment]
Oops, no. I think this can be merged now.

**Comment by MikeInnes** [comment]
Alright, thanks for setting this up!

**Pull request merged**